### ISSUE TITLE
pyais v2

### ISSUE THREAD

**Issue by ambroisernd** [M0r13n/pyais#71]
Hello,

what is is projet : https://pypi.org/project/pyais-2/#description ? It implements NMEASorter in message.py (not very robust tho)

Ambroise

**Comment by M0r13n** [comment]
what?

**Comment by ambroisernd** [comment]
`class NMEASorter:

    def __init__(self, messages: typing.Iterable[bytes]):
        self.unordered = messages

    def __iter__(self) -> typing.Generator[bytes, None, None]:
        buffer: typing.Dict[typing.Tuple[int, bytes], typing.List[typing.Optional[bytes]]] = {}

        for msg in self.unordered:
            # decode nmea header

            parts = msg.split(b',')
            if len(parts) < 5:
                raise InvalidNMEAMessageException("Too few message parts")

            try:
                frag_cnt = int(parts[1])
                frag_num = int(parts[2]) - 1
                seq_id = int(parts[3]) if parts[3] else 0
                channel = parts[4]
            except ValueError as e:
                raise InvalidNMEAMessageException() from e

            if frag_cnt > 20:
                raise InvalidNMEAMessageException("Frag count is too large")

            if frag_num >= frag_cnt:
                raise InvalidNMEAMessageException("Fragment number greater than Fragment count")

            if frag_cnt == 1:
                # A sentence with a fragment count of 1 is complete in itself
                yield msg
                continue

            # seq_id and channel make a unique stream
            slot = (seq_id, channel)

            if slot not in buffer:
                buffer[slot] = [None, ] * frag_cnt

            buffer[slot][frag_num] = msg
            msg_parts = buffer[slot][0:frag_cnt]
            if all([m is not None for m in msg_parts]):
                yield from msg_parts  # type: ignore
                del buffer[slot]

        # yield all remaining messages that were not fully decoded
        for msg_parts in buffer.values():
            yield from filter(lambda x: x is not None, msg_parts)  # type: ignore`

`slot = (seq_id, channel)` if we have two message:
eg : !AIVDM,3,1,4,A and !AIVDM,2,2,4,A they will have the same slot: (4, A) but they should not be parsed together.

I can't reproduce this bug using pip install pyais (version on this github), but it appears using pip install pyais2

**Comment by M0r13n** [comment]
I think that I understand. 

pyais was a experiment of mine to work around breaking changes in pyais. But I ultimately decided to just release version 2.0.0 of pyais. But I never actually deleted `pyais-2`. I have now made up for this and deleted this project from PyPi.

The `NMEASorter` was present in `pyais` until version 2.0.0. I removed this class because it was not working correctly.

In current versions `pyais` this functionality is available to all classes that inherit from [AssembleMessages](https://github.com/M0r13n/pyais/blob/a376ab46411afc23f49895345d14f570d59689e1/pyais/stream.py#L47)
